src/zocalo/cli/configure-rabbitmq.py

```diff
@@ -0,0 +1,284 @@
+        raise NotImplementedError(f"Component {component} not recognised")
+
+    @create_component.register
+    def _(self, binding: BindingSpec):
+        self.binding_declare(binding)
+
+    @delete_component.register
+    def _(self, binding: BindingSpec):
+        self.bindings_delete(
+            vhost=binding.vhost,
+            source=binding.source,
+            destination=binding.destination,
+            destination_type=binding.destination_type,
+            properties_key=binding.properties_key,
+        )
+
+    @create_component.register
+    def _(self, exchange: ExchangeSpec):
+        self.exchange_declare(exchange)
+
+    @delete_component.register
+    def _(self, exchange: ExchangeSpec, **kwargs):
+        self.exchange_delete(vhost=exchange.vhost, name=exchange.name, **kwargs)
+
+    @create_component.register
+    def _(self, policy: PolicySpec):
+        self.set_policy(policy)
+
+    @delete_component.register
+    def _(self, policy: PolicySpec):
+        self.clear_policy(vhost=policy.vhost, name=policy.name)
+
+    @create_component.register
+    def _(self, queue: QueueSpec):
+        self.queue_declare(queue)
+
+    @delete_component.register
+    def _(self, queue: QueueSpec, if_unused: bool = False, if_empty: bool = False):
+        self.queue_delete(
+            vhost=queue.vhost, name=queue.name, if_unused=if_unused, if_empty=if_empty
+        )
+
+    @create_component.register
+    def _(self, user: UserSpec):
+        self.add_user(user)
+
+    @delete_component.register
+    def _(self, user: UserSpec):
+        self.delete_user(name=user.name)
+
+
+def update_config(api: RabbitMQAPI, incoming, current):
+    cls = type(incoming[0])
+    current = [cls(**c.dict()) for c in current]
+    for cc in current:
+        if cc in incoming:
+            if hasattr(cc, "name"):
+                logger.debug(f"{cls.__name__} {cc.name} already exists")
+            elif hasattr(cc, "source"):
+                logger.debug(
+                    f"{cls.__name__} {cc.source or 'default'}->{cc.destination} already exists"
+                )
+        else:
+            if hasattr(cc, "name"):
+                if cc.name == "" or "amq." in cc.name:
+                    continue
+            if hasattr(cc, "source"):
+                if cc.source == "" or "amq." in cc.source:
+                    continue
+            if getattr(cc, "auto_delete") or getattr(cc, "exclusive"):
+                continue
+            if isinstance(cc, BindingSpec) and cc["source"] == "":
+                continue
+            logger.info(f"deleting {cls.__name__} {cc}")
+            api.delete_component(cc)
+    for ic in incoming:
+        if ic not in current:
+            logger.info(f"creating {cls.__name__} {ic}")
+            api.create_component(ic)
+
+
+def hash_password(passwd: str) -> str:
+    salt = os.urandom(4)
+    utf8 = passwd.encode("utf-8")
+    temp1 = salt + utf8
+    temp2 = hashlib.sha256(temp1).digest()
```

Comment:
```suggestion
    temp2 = hashlib.sha256(temp1).digest()  # lgtm
```
The LGTM crypto warning is correct, but also nothing we can do about it. Other than adding a comment to the offending line.